src/System.Net.WebHeaderCollection/tests/WebHeaderCollectionTest.cs

```diff
@@ -0,0 +1,236 @@
+        {
+            WebHeaderCollection w = new WebHeaderCollection();
+            Assert.Throws<ArgumentNullException>(() => w[""] = "value");
+        }
+
+        [Fact]
+        public void RequestThenResponseHeaders_Add_Fail()
+        {
+            WebHeaderCollection w = new WebHeaderCollection();
+            w[HttpRequestHeader.Accept] = "text/json";
+            Assert.Throws<InvalidOperationException>(() => w[HttpResponseHeader.ContentLength] = "123");
+        }
+
+        [Fact]
+        public void ResponseThenRequestHeaders_Add_Fail()
+        {
+            WebHeaderCollection w = new WebHeaderCollection();
+            w[HttpResponseHeader.ContentLength] = "123";
+            Assert.Throws<InvalidOperationException>(() => w[HttpRequestHeader.Accept] = "text/json");
+        }
+
+        [Fact]
+        public void ResponseHeader_QueryRequest_Fail()
+        {
+            WebHeaderCollection w = new WebHeaderCollection();
+            w[HttpResponseHeader.ContentLength] = "123";
+            Assert.Throws<InvalidOperationException>(() => w[HttpRequestHeader.Accept]);
+        }
+
+        [Fact]
+        public void RequestHeader_QueryResponse_Fail()
+        {
+            WebHeaderCollection w = new WebHeaderCollection();
+            w[HttpRequestHeader.Accept] = "text/json";
+            Assert.Throws<InvalidOperationException>(() => w[HttpResponseHeader.ContentLength]);
+        }
+
+        [Fact]
+        public void NameValue_Add_Success()
+        {
+            WebHeaderCollection w = new WebHeaderCollection();
+            w.Add("Accept", "text/json");
+        }
+
+        [Fact]
+        public void NameValue_AddNullName_Fail()
+        {
+            WebHeaderCollection w = new WebHeaderCollection();
+            Assert.Throws<ArgumentNullException>(() => w.Add(null, "test"));
+        }
+
+        [Fact]
+        public void NameValue_AddEmptyName_Fail()
+        {
+            WebHeaderCollection w = new WebHeaderCollection();
+            Assert.Throws<ArgumentException>(() => w.Add("", "test"));
+        }
+
+        [Fact]
+        public void NameValue_AddEmptyValue_Success()
+        {
+            WebHeaderCollection w = new WebHeaderCollection();
+            w.Add("custom", "");
+        }
+
+        public static object[][] InvalidNames = {
+            new object[] { "(" },
+            new object[] { "\u1234" },
+            new object[] { "\u0019" }
+        };
+
+        [Theory, MemberData("InvalidNames")]
+        public void CheckBadChars_InvalidName(string name)
+        {
+            WebHeaderCollection w = new WebHeaderCollection();
+            Assert.Throws<ArgumentException>(() => w["("] = "test");
+        }
+
+        public static object[][] InvalidValues = {
+            new object[] { "value1\rvalue2\r" },
+            new object[] { "value1\nvalue2\r" },
+            new object[] { "value1\u007fvalue2" },
+            new object[] { "value1\u0019value2" },
```

Comment:
I'm not aware of any workarounds other than something that avoids the problematic input being passed into the method, e.g. just removing that test case if it's not important, changing to putting it in its own [Fact], changing the whole theory to just be a [Fact] that loops through the inputs in a normal loop, etc.